marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java

```diff
@@ -405,16 +405,14 @@ private RunStepResponse runIngester(RunStepResponse runStepResponse, Collection<
             runStepResponse.withStatus(JobStatus.COMPLETED_PREFIX + step);
             JsonNode jobDoc;
             try {
-                jobDoc = jobDocManager.postJobs(jobId, JobStatus.COMPLETED_PREFIX + step, step, step, runStepResponse);
-            }
-            catch (Exception e) {
-                throw e;
-            }
-            try {
+                try {
+                    jobDoc = jobDocManager.postJobs(jobId, JobStatus.COMPLETED_PREFIX + step, step, step, runStepResponse);
+                } catch (Exception e) {
+                    throw e;
+                }
                 return StepRunnerUtil.getResponse(jobDoc, step);
-            }
-            catch (Exception ex)
-            {
+            } catch (Exception ex) {
+            } finally {
                 return runStepResponse;
```

Comment:
Why is return runStepResponse moved to finally block here ?

Reply:
i saw the stepResponses missing some value like flowName for RunFlowResponse object for the case. It is just data completion problem when i debugged.